src/mono/System.Private.CoreLib/src/System/Reflection/CustomAttribute.cs

```diff
@@ -306,6 +303,9 @@ internal static object[] GetCustomAttributes(ICustomAttributeProvider obj, bool
         {
             ArgumentNullException.ThrowIfNull(obj);
 
+            if (IsUserCattrProvider(obj))
+                return obj.GetCustomAttributes(typeof(Attribute), inherit);
```

Comment:
I am passing in `typeof(Attribute)` here to match CoreCLR implementation: https://github.com/dotnet/runtime/blob/c07cb4441d6270327472e88919f3d4eb6cc0450f/src/coreclr/System.Private.CoreLib/src/System/Attribute.CoreCLR.cs#L555-L567